### PR TITLE
add ruby tests 3.0/3.1

### PR DESCRIPTION
**Pull request by jayashrig158**
add ruby tests 3.0/3.1

Signed-off-by: jayashri garud <jgarud@msystechnologies.com>

### Description

[Please describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussions that are relevant]

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -11,6 +11,6 @@ end
 group :debug do
   gem "pry"
   gem "pry-byebug"
-  gem "pry-stack_explorer"
+  gem "pry-stack_explorer", "~> 0.4.13" # pin until we support ruby 2.5
```
Comment (jeremiahsnapp) [inline Gemfile:14]: Do we need to pin pry-stack_explorer?

File: .expeditor/verify.pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+- label: run-specs-windows
+  command:
+    - bundle config --local path vendor/bundle
+    - bundle install --jobs=7 --retry=3
+    - bundle exec rake
+  expeditor:
+    executor:
+      docker:
+        host_os: windows
```
Comment (jeremiahsnapp) [inline .expeditor/verify.pipeline.yml:62]: Now you can use the rubydistros windows image with ruby 3.1 installed for the test on windows.

```
- label: run-specs-ruby-3.1-windows
  command:
    - bundle config --local path vendor/bundle
    - bundle install --jobs=7 --retry=3
    - bundle exec rake
  expeditor:
    executor:
      docker:
        host_os: windows
        shell: ["powershell", "-Command"]
        image: rubydistros/windows-2019:3.1
```

**Review by jeremiahsnapp — CHANGES_REQUESTED** [review]
Where is the content for the .expeditor/run_linux_tests.sh script?

File: Gemfile (hunk 1 above)
Comment (jayashrig158) [reply Gemfile:14]: Yes @jeremiahsnapp latest pry-stack-explorer doesn't support ruby 2.5

**Pull request merged**